streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/HighAvailabilityTaskAssignor.java

```diff
@@ -337,14 +342,11 @@ private int tasksPerThread(final UUID client, final List<Map<UUID, List<TaskId>>
      * Wraps a priority queue of clients and returns the next valid candidate(s) based on the current task assignment
      */
     static class ValidClientsByTaskLoadQueue<UUID> {
-        private final int numClientsPerTask;
         private final PriorityQueue<UUID> clientsByTaskLoad;
         private final List<Map<UUID, List<TaskId>>> allStatefulTaskAssignments;
 
-        ValidClientsByTaskLoadQueue(final int numClientsPerTask,
-                                      final PriorityQueue<UUID> clientsByTaskLoad,
-                                      final List<Map<UUID, List<TaskId>>> allStatefulTaskAssignments) {
-            this.numClientsPerTask = numClientsPerTask;
```

Comment:
Since we may now "use up" some standby replicas during the warmup replica assignment, we no longer want a constant N clients returned here. The number of remaining standbys to assign is now a parameter of its `poll`